ENIGMAsystem/SHELL/Widget_Systems/widgets_mandatory.h

```diff
@@ -22,22 +22,48 @@
 #include "libEGMstd.h"
 
 #include <string>
-using std::string;
 
-namespace enigma
-{
+#define DEBUG_MESSAGE(msg, severity) enigma_user::show_debug_message((std::string)msg + " | " + __FILE__ + ":" + std::to_string(__LINE__), severity)
```

Comment:
Wrapping msg in parentheses will show you places where you're adding `char*`s. Incidentally, you don't need the + between `" | "`, `__FILE__` and `":"`, as all of them are string literals.